component/camera.js

```diff
@@ -13,12 +14,35 @@ class Camera {
 
     captureImage() {
         return new Promise((resolve, reject) => {
-            NodeWebcam.capture("test_picture", this.opts, function (err, data) {
+        
+            if (config.camType == "USB"){
+                NodeWebcam.capture("test_picture", this.opts, function (err, data) {
                 if (err) {
                     reject(err);
                 }
                 resolve(data);
+                });
+            }
+            if (config.camType == "ONVIF"){
+            console.log(`[DEBUG] Starting image capture using: ${config.xaddrOnvif}`)
+        
+            // Create an OnvifDevice object
+            let device = new onvif.OnvifDevice({
+            xaddr: config.xaddrOnvif,
+            user : config.userOnvif,
+            pass : config.passOnvif
+            });
+            
+            // Initialize the OnvifDevice object
+            device.init().then(() => {
+            // Get the data of the snapshot
+            console.log('[DEBUG] Fetching the snapshot...');
+            return device.fetchSnapshot();
+            }).then((res) => {
+                console.log('[DEBUG] Done!');
+                resolve(res.body);
             });
```

Comment:
Op zich prima dit, maar architectuur technisch wil je dit dus anders doen. Je kunt twee verschillende Camera subtypes maken, gedeelde code zit dan in Camera en de specifieke code doen we dan in UsbCamera en OnvifCamera classes. Dan kun je afhankelijk van configuratie of beschikbaarheid de juiste class inladen.